### PR TITLE
springserve adapter

### PR DESCRIPTION
**Pull request by JohnBuonomo**
adapter for the springserve bidder

### CODE REVIEW

**Comment by mkendall07** [comment]
Thanks for the PR @JohnBuonomo 

Can you please submit your params object needed to wire up the adapter? Example
https://github.com/prebid/Prebid.js/blob/master/integrationExamples/gpt/pbjs_example_gpt.html#L97-L100

Also please include actual test params that can be used to retrieve an ad for validation. Thanks very much!

**Comment by JohnBuonomo** [comment]
@mkendall07  The testing params are included in that last commit. Let me know if something else is needed. -John

**Comment by protonate** [comment]
Hi @JohnBuonomo, the placement size info is already set on the adUnit object [here](https://github.com/yellowhammer/Prebid.js/blob/master/integrationExamples/gpt/pbjs_example_gpt.html#L63), so you can remove the height and width properties from the springserve bid object.

**Comment by JohnBuonomo** [comment]
@protonate I made the requested changes to the params.

**Comment by protonate** [comment]
Thanks for the changes, @JohnBuonomo. The height and width are undefined when set in `handleSpringServeCB()`, you can pull them from `bidRequest.sizes` however. I'll comment inline.

File: src/adapters/springserve.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        var responseBid = responseObj.seatbid[0].bid[0];
+        var requestObj = bidmanager.getPlacementIdByCBIdentifer(responseBid.impid);
+        var bid = bidfactory.createBid(1);
+
+        //assign properties from the original request to the bid object
+        for (var i = 0; i < requestObj.bids.length; i++ ){
+          var bidRequest = requestObj.bids[i];
+          if(bidRequest.bidder = "springserve"){
+             var placementCode = bidRequest.placementCode;
+             bid.width = bidRequest.params.width;
```
Comment (protonate) [inline src/adapters/springserve.js]: Width is undefined.

File: src/adapters/springserve.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+             bid.height = bidRequest.params.height;
```
Comment (protonate) [inline src/adapters/springserve.js]: Height is undefined.

**Comment by protonate** [comment]
Thanks for the fixes. Please squash commits and we can get this merged. Thanks!

**Comment by JohnBuonomo** [comment]
Squashed. Thank you!

File: src/adapters/springserve.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            responseObj.seatbid[0].bid[0] !== undefined){
+            //look up the request attributs stored in the bidmanager
+            var responseBid = responseObj.seatbid[0].bid[0];
+            var requestObj = bidmanager.getPlacementIdByCBIdentifer(responseBid.impid);
+            var bid = bidfactory.createBid(1);
+
+            //assign properties from the original request to the bid object
+            for (var i = 0; i < requestObj.bids.length; i++ ){
+                var bidRequest = requestObj.bids[i];
+                if(bidRequest.bidder = "springserve"){
```
Comment (protonate) [inline src/adapters/springserve.js]: @JohnBuonomo I noticed this while applying code style tools, but I expect you want this to be an equality check and not assignment. I'll put this fix into the upcoming code style PR.

Comment (JohnBuonomo) [inline src/adapters/springserve.js]: @protonate Ugh.  Yes. Thank you for catching that.

**Pull request merged**